platform-hub-api/app/services/kubernetes/changeset_service.rb

```diff
@@ -2,29 +2,37 @@ module Kubernetes
   module ChangesetService
     extend self
 
-    CHANGSET_ACTIONS = [
-      :update_kubernetes_identity,
-      :revoke_kubernetes_token,
-      :claim_kubernetes_token,
-      :update_kubernetes_robot_token,
-      :destroy_kubernetes_robot_token,
-      :escalate_kubernetes_token,
-      :deescalate_kubernetes_token
-    ]
-
-    def get_events(cluster, since)
-      audit_entities(CHANGSET_ACTIONS, cluster).where("created_at > ?", since)
-    end
+    AUDITABLE_CHANGESET_ACTIONS = {
+      'KubernetesToken' => [
+        :create,
+        :update,
+        :destroy,
+        :escalate,
+        :deescalate,
+      ]
+    }
 
-    def last_sync(cluster)
-      audit_entities(:sync_kubernetes_tokens, cluster).first.try(:created_at) || 1.year.ago.utc.to_s(:db)
+    def get_events(cluster)
+      since = last_sync(cluster)
+
+      AUDITABLE_CHANGESET_ACTIONS.collect do |auditable_type, actions|
+        audit_entities_by_cluster_and_auditable_type(cluster, auditable_type, actions).where("created_at > ?", since)
+      end.flatten
     end
 
     private
 
-    def audit_entities(actions, cluster)
+    def last_sync(cluster)
+      audit_entities_by_cluster(cluster, :sync_kubernetes_tokens).first.try(:created_at) || 1.year.ago.utc.to_s(:db)
+    end
```

Comment:
Given now that we automatically sync behind the scenes, the current changeset entries can be a bit misleading. I wonder if we should rethink this bit – we could just display all changeset entries for the last X days **and** also show the `sync_kubernetes_tokens` event (highlighted) in the UI, so it becomes more like a history page.

Don't have to do this now! We can do this in a separate PR, or maybe as part of a new Audit UI – just wanted to get your thoughts on it at this stage.

Reply:
Let's discuss and make changes in a separate PR.

Reply:
Cool 👍